testing/src/main/java/io/stargate/it/http/docsapi/JsonSchemaResourceIntTest.java

```diff
@@ -0,0 +1,144 @@
+    results.forEach(row -> ids.add(String.format("'%s'", row.getString("key"))));
+
+    // This resets the JSON schema on the table
+    session.execute(String.format("ALTER TABLE %s WITH COMMENT = ''", TARGET_COLLECTION));
+
+    if (!ids.isEmpty()) {
+      session.execute(
+          String.format(
+              "DELETE FROM %s WHERE key IN (%s)", TARGET_COLLECTION, String.join(", ", ids)));
+    }
+  }
+
+  private static void initAuth() throws IOException {
+    OBJECT_MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    RequestBody requestBody =
+        RequestBody.create(
+            MediaType.parse("application/json"),
+            OBJECT_MAPPER.writeValueAsString(new Credentials("cassandra", "cassandra")));
+
+    Request request =
+        new Request.Builder()
+            .url(String.format("%s:8081/v1/auth/token/generate", host))
+            .post(requestBody)
+            .addHeader("X-Cassandra-Request-Id", "foo")
+            .build();
+    Response response = CLIENT.newCall(request).execute();
+    ResponseBody body = response.body();
+
+    assertThat(body).isNotNull();
+    AuthTokenResponse authTokenResponse =
+        OBJECT_MAPPER.readValue(body.string(), AuthTokenResponse.class);
+    authToken = authTokenResponse.getAuthToken();
+    assertThat(authToken).isNotNull();
+  }
+
+  @Test
+  public void testIt() throws IOException {
+    RestUtils.post(
+        authToken,
+        hostWithPort + "/v2/namespaces/" + keyspace + "/collections",
+        "{\"name\":\"collection\"}",
+        201);
+    JsonNode schema =
+        OBJECT_MAPPER.readTree(this.getClass().getClassLoader().getResource("schema.json"));
+    RestUtils.put(authToken, collectionPath + "/json-schema", schema.toString(), 200);
+
+    String schemaResp = RestUtils.get(authToken, collectionPath + "/json-schema", 200);
+    assertThat(schema.toString())
+        .isEqualTo(OBJECT_MAPPER.readTree(schemaResp).requiredAt("/schema").toString());
+
+    JsonNode obj = OBJECT_MAPPER.readTree("{\"id\":1, \"name\":\"a\", \"price\":1}");
+    RestUtils.put(authToken, collectionPath + "/1", obj.toString(), 200);
+
+    obj = OBJECT_MAPPER.readTree("{\"id\":1, \"price\":1}");
+    String resp = RestUtils.put(authToken, collectionPath + "/1", obj.toString(), 400);
+    JsonNode data = OBJECT_MAPPER.readTree(resp);
+    assertThat(data.requiredAt("/description").textValue())
+        .isEqualTo("Invalid JSON: [object has missing required properties ([\"name\"])]");
+
+    obj = OBJECT_MAPPER.readTree("{\"id\":1, \"name\":\"a\", \"price\":-1}");
+    resp = RestUtils.put(authToken, collectionPath + "/1", obj.toString(), 400);
+    data = OBJECT_MAPPER.readTree(resp);
+    assertThat(data.requiredAt("/description").textValue())
+        .isEqualTo(
+            "Invalid JSON: [numeric instance is lower than the required minimum (minimum: 0, found: -1)]");
+  }
+
+  @Test
+  public void testInvalidSchema() throws IOException {
```

Comment:
What about a test for getting a schema from a collection that doesn't have one?

Reply:
Good point +1